dpcpp/solver/batch_cg_kernels.dp.cpp

```diff
@@ -43,12 +43,183 @@ namespace batch_cg {
 #include "dpcpp/matrix/batch_csr_kernels.hpp.inc"
 #include "dpcpp/matrix/batch_dense_kernels.hpp.inc"
 #include "dpcpp/matrix/batch_ell_kernels.hpp.inc"
+#include "dpcpp/solver/batch_cg_kernels.hpp.inc"
 
 
 template <typename T>
 using settings = gko::kernels::batch_cg::settings<T>;
 
 
+__dpct_inline__ int get_group_size(int value,
+                                   int subgroup_size = config::warp_size)
+{
+    int num_sg = ceildiv(value, subgroup_size);
+    return num_sg * subgroup_size;
+}
+
+
+template <typename ValueType>
+class kernel_caller {
+public:
+    kernel_caller(std::shared_ptr<const DefaultExecutor> exec,
+                  const settings<remove_complex<ValueType>> settings)
+        : exec_{std::move(exec)}, settings_{settings}
+    {}
+
+    template <typename StopType, const int subgroup_size,
+              const int n_shared_total, typename PrecType, typename LogType,
+              typename BatchMatrixType>
+    __dpct_inline__ void launch_apply_kernel(
```

Comment:
```suggestion
    void launch_apply_kernel(
```
it is not the device code